requests/dialer.go

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (C) 2020 The "MysteriumNetwork/node" Authors.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package requests
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"time"
+)
+
+// Dialer wraps default go dialer with extra features.
+type Dialer struct {
+	zeroDialer *net.Dialer
```

Comment:
Do we need to have a `zero` prefix for the dialer? 
```suggestion
	dialer *net.Dialer
```